pkg/tfbridge/detailed_diff.go

```diff
@@ -259,6 +292,78 @@ func (differ detailedDiffer) makeListDiff(
 	return diff
 }
 
+type setChangeIndex struct {
+	engineIndex   int
+	newStateIndex int
+	oldChanged    bool
+	newChanged    bool
+}
+
+func (differ detailedDiffer) makeSetDiff(
+	path propertyPath, old, new resource.PropertyValue,
+) map[detailedDiffKey]*pulumirpc.PropertyDiff {
+	diff := make(map[detailedDiffKey]*pulumirpc.PropertyDiff)
+	oldList := old.ArrayValue()
+	newList := new.ArrayValue()
+	newInputsList := []resource.PropertyValue{}
```

Comment:
The testing is good so this is probably correct, but I've been spending way too much time reading into it trying to figure out if things line up...

Consider a nit but if you would like to play some code golf, code this be broken down for clarity? It would also make testing separable.

First phase:

```go
type setDiff struct {
   added []PropertyValue
   removed []PropertyValue
}

func (differ detailedDiff) computeSetDiff(path propertyPath, old, new []resource.PropertyValue) setDiff
```

In this phase Terraform hashes are uses to decide if two elements are considered equal or not.

Second phase:

```go
func (differ detailedDiffer) formatPropertyDiff(setDiff, etc) map[detailedDiffKey]*pulumirpc.PropertyDiff {
}
```

In this second phase we are looking up the engine-compatible index for every added or removed resource.PropertyValue. We are using this by comparing PropertyValue for equality, not for their "Terraform" equality. So it should not be using differ.calculateSetHashIndexMap or any some such, correct? IF we cannot find the exact PropertyValue match in what the engine will be looking for, then we cannot display this properly, and it's time to spill over to the warning system and display that there.